esp/src/src-react/components/GetLogicalFilePart.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from "react";
+import { DefaultButton, DetailsList, DetailsListLayoutMode, IColumn } from "@fluentui/react";
+import { SizeMe } from "react-sizeme";
+import { csvParse } from "d3-dsv";
+import { DaliService, WsDali } from "@hpcc-js/comms";
+import { scopedLogger } from "@hpcc-js/util";
+import { TableGroup } from "./forms/Groups";
+import nlsHPCC from "src/nlsHPCC";
+import { HolyGrail } from "../layouts/HolyGrail";
+
+class MyDaliService extends DaliService {
+    GetLogicalFilePart(request: WsDali.GetLogicalFilePartRequest): Promise<WsDali.ResultResponse> {
+        return this._connection.send("GetLogicalFilePart", request, "json", false, undefined, "ResultResponse");
+    }
+}
+
+const logger = scopedLogger("src-react/components/GetLogicalFilePart.tsx");
+
+const myDaliService = new MyDaliService({ baseUrl: "" });
+
+interface GetLogicalFilePartProps {
+}
+
+export const GetLogicalFilePart: React.FunctionComponent<GetLogicalFilePartProps> = ({
+
+}) => {
+
+    const [columns, setColumns] = React.useState<IColumn[]>([]);
+    const [items, setItems] = React.useState<any[]>([]);
+    const [fileName, setFileName] = React.useState<string>("");
+    const [partNumber, setPartNumber] = React.useState<number>(1);
+
+    const onSubmit = React.useCallback(() => {
+        myDaliService.GetLogicalFilePart({ FileName: fileName, PartNumber: partNumber }).then(response => {
+            const data = csvParse(response.Result);
+            setColumns(data.columns.map((col, idx) => {
+                return {
+                    key: col,
+                    name: col,
+                    fieldName: col,
+                    minWidth: 100
+                };
+            }));
+            setItems(data);
+        }).catch(err => logger.error(err));
+    }, [fileName, partNumber]);
+
+    return <HolyGrail
+        header={<span><TableGroup fields={{
+            "FileName": { label: nlsHPCC.FileName, type: "string", value: fileName },
+            "PartNumber": { label: nlsHPCC.PartNumber, type: "number", value: partNumber },
+        }} onChange={(id, value) => {
+            switch (id) {
+                case "fileName":
```

Comment:
This problem persists. The casing here should match the keys in the fields object (lines 50 & 51), eg: FileName and PartNumber. I wouldn't think this is the case, but perhaps it's a difference between OSX and Windows? But generally speaking, you can see whether or not the onChange callback function is updating the field correctly when viewing the page.